version: "2.3"
services:
  challengefile:
    build: .
    ports: 
      - "8080:8080"
    networks: 
      - docker_network
    links:
      - postgres:postgres
    environment:
      DB_HOSTNAME: 'postgres_container'
      DB_PORT: '5432'
      DB_USER: 'postgres'
      DB_PASSWORD: '1234'
      DB_DATABASE: 'store'
    volumes:
      - .:/go/src/filechallenge
    restart: on-failure
    depends_on: 
      - "postgres"
  postgres:
    restart: always
    image: postgres
    container_name: "postgres_container"
    ports:
      - "5432:5432"
    networks:
      - docker_network
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: 1234
      POSTGRES_DB: store
    volumes:
      - pgvol:/var/lib/postgresql/data
  pgAdmin:
    restart: always
    image: dpage/pgadmin4
    container_name: pgadmin
    ports:
      - "8000:80"
    environment:
      PGADMIN_DEFAULT_EMAIL: 1234@admin.com
      PGADMIN_DEFAULT_PASSWORD: 1234
    volumes:
      - pgadmin:/var/lib/pgadmin
    networks:
    - docker_network
networks:
  docker_network:
    driver: bridge
 
volumes:
  pgvol:
  pgadmin: